ui/apps/platform/src/Containers/VulnMgmt/Entity/Policy/VulnMgmtPolicyOverview.js

```diff
@@ -216,26 +214,70 @@ const VulnMgmtPolicyOverview = ({ data, entityContext, setRefreshTrigger }) => {
+                    }
+                }
+            }
+            ${fragmentToUse}
+        `;
+
+        const fullEntityContext = workflowState.getEntityContext();
+        const failingDeploymentQueryVariables = {
+            id,
+            deploymentQuery: queryService.objectToWhereClause({
+                'Policy Violated': true,
+                ...queryService.entityContextToQueryObject(fullEntityContext),
+            }),
+            ...vulMgmtPolicyQuery,
+            pagination: queryService.getPagination(sort, page, 2 /* TODO Remove this param */),
+        };
+
         policyFindingsContent = (
-            <div className="pdf-page pdf-stretch pdf-new flex shadow rounded relativebg-base-100 mb-4 mx-4">
-                <TableWidget
-                    header={`${failingDeployments.length} ${pluralize(
+            <Query query={failingDeploymentQuery} variables={failingDeploymentQueryVariables}>
+                {({ loading, data: deploymentData, previousData }) => {
+                    const policyData = deploymentData?.policy || previousData?.policy;
+                    if (loading && !policyData) {
+                        return <Loader />;
+                    }
+                    const failingDeployments = policyData.deployments;
+                    const totalCount = policyData.deploymentCount;
+                    const headerText = `${totalCount} ${pluralize(
                         entityTypes.DEPLOYMENT,
-                        failingDeployments.length
-                    )} ${pluralizeHas(failingDeployments.length)} failed across this policy`}
-                    rows={failingDeployments}
-                    entityType={entityTypes.DEPLOYMENT}
-                    noDataText="No deployments have failed across this policy"
-                    className="bg-base-100"
-                    columns={getDeploymentTableColumns(workflowState)}
-                    defaultSorted={[
-                        {
-                            id: 'priority',
-                            desc: false,
-                        },
-                    ]}
-                />
-            </div>
+                        totalCount
+                    )} ${pluralizeHas(totalCount)} failed across this policy`;
+                    return (
+                        <div className="pdf-page pdf-stretch pdf-new flex shadow rounded relative bg-base-100 mb-4 mx-4">
+                            <TableWidget
+                                header={headerText}
+                                rows={failingDeployments}
+                                entityType={entityTypes.DEPLOYMENT}
+                                noDataText="No deployments have failed across this policy"
+                                className="bg-base-100"
+                                columns={getDeploymentTableColumns(workflowState)}
+                                /* TODO Remove `pageSize` */
+                                pageSize={2}
```

Comment:
I've hard coded a page size of 2 for easier testing, but if this approach seems OK I'll remove this and test the component at actual scale.